### PR TITLE
docs: code samples for dataframe.any, dataframe.all and dataframe.prod

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

dataframe.any: https://screenshot.googleplex.com/3tDhwR9uLRpEZMj
dataframe.all: https://screenshot.googleplex.com/BmCTG4v88CKSzSD
dataframe.prod: https://screenshot.googleplex.com/4yjrcDP72n757hR

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 1]
```diff
@@ -2620,8 +2674,31 @@ def prod(self, axis=0, *, numeric_only: bool = False):
...
+            >>> df
+
+        Calculating the product of each column(the default).
+
+            >>> df.prod()
+            A        6.0
+            B    160.875
+            dtype: Float64
+
+        Calculating the product along the rows.
```
Comment (shobsi) [inline third_party/bigframes_vendored/pandas/core/frame.py:2692]: [nit] not sure if "along" would be very clear, can we say "Calculating the product of each row" to be coherent with the previous sentence?

**Review by ashleyxuu — COMMENTED** [review]
Could you also do a staging page like: https://github.com/googleapis/python-bigquery-dataframes/pull/212#issue-1996123354

You can follow: https://g3doc.corp.google.com/devrel/cloud/infra/devrel_services/g3doc/dev/content_automation/cloud_rad/python.md?cl=head#staging-local-changes-from-github-to-devsite

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 2]
```diff
@@ -2620,8 +2674,31 @@ def prod(self, axis=0, *, numeric_only: bool = False):
...
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> df = bpd.DataFrame({"A": [1, 2, 3], "B": [4.5, 5.5, 6.5]})
+            >>> df
+
+        Calculating the product of each column(the default).
```
[the remark this answers is not in the document]
Comment (Genesis929) [reply third_party/bigframes_vendored/pandas/core/frame.py]: Done.

File: third_party/bigframes_vendored/pandas/core/frame.py (hunk 1 above)
Comment (Genesis929) [reply third_party/bigframes_vendored/pandas/core/frame.py]: Done.

**Review by shobsi — APPROVED** [review]

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**